cloudferrylib/os/actions/verify_failed_instances_deleted.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+
+from cloudferrylib.utils import utils
+from cloudferrylib.base.action import action
+
+LOG = utils.get_log(__name__)
+
+COMPUTE = 'compute'
+
+
+class VerifyFailedInstancesDeleted(action.Action):
+    """ Verify that all instances which failed to create were deleted. """
+    def run(self, **kwargs):
+        dst_compute = self.cloud.resources[COMPUTE]
+        failed_instances = dst_compute.failed_instances
+        if failed_instances:
+            LOG.warning('Instances failed to create and were not deleted: %s',
```

Comment:
Please change message to something more meaningful for migration operator, I doubt this can be understood by anyone not familiar with details of your patch.